konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/heatmap/DrawHeatmapStep.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  ******************************************************************************
+ *  * Copyright (c) 2020 Konduit K.K.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+
+package ai.konduit.serving.data.image.step.point.heatmap;
+
+import ai.konduit.serving.annotation.json.JsonName;
+import ai.konduit.serving.pipeline.api.step.PipelineStep;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Singular;
+import lombok.experimental.Accessors;
+
+import java.util.List;
+
+/**
+ * Draw a heatmap using 2D points.<br>
+ * Heat will accumulate over the lifecycle of this step.<br>
+ * Configuration:<br>
+ * <ul>
+ *     <li><b>points</b>: Names of the points to be used for the heatmap. Accepts both single points and lists of points.
+ *     <li><b>radius</b>: Size of area influenced by a point.</li>
+ *     <li><b>fadingFactor</b>: Optional. Value between 0 and 1. 0: No Fade, 1: Instant fade</li>
+ *     <li><b>image</b>: Optional. Name of the image to use as size reference</li>
+ *     <li><b>width</b>: Must be provided when <b>image</b> isn't set. Used to resolve position of points with relative addressing (dimensions between 0 and 1)</li>
+ *     <li><b>height</b>: Must be provided when <b>image</b> isn't set. Used to resolve position of points with relative addressing (dimensions between 0 and 1)</li>
+ *     <li><b>outputName</b>: Name of the output image</li>
+ * </ul>
+ *
+ * @author Paul Dubs
+ */
+@Builder
+@Data
+@Accessors(fluent = true)
+@AllArgsConstructor
+@JsonName("DRAW_HEATMAP")
+@Schema(description = "A pipeline step that configures how to draw 2D points, optionally on an image.")
+public class DrawHeatmapStep implements PipelineStep {
```

Comment:
Again, "on image" vs. "image as size reference".
Being able to overlay on an image would definitely be a nice addition (at the very least, we'll open an issue and add it later)